### ISSUE TITLE
Allow creating alerts and filtering by team

### ISSUE THREAD

**Issue by johnrudell** [getsentry/sentry#65723]
### Problem Statement

When I create an alert, I can add filters for the event, but the field "team" is not supported. You can add your team to the "Establish ownership" step, but this only provides permissions for the specified team to edit this alert.

### Solution Brainstorm

It would be great if we could create alerts and filter the events by team.

### Product Area

Alerts

**Comment by johnrudell** [comment]
@KobbyBawuah I believe this is correct. There are issues under "Related Issues" for my alert with the team filter. It seems like `is` is the only supported filter key, but in reality this can be expanded to include other filters.

**Comment by Dhrumil-Sentry** [comment]
@johnrudell have you considered using auto assignment which allows you to use the following filter:
![image](https://github.com/getsentry/sentry/assets/109983869/1c00fa08-ad23-409a-a7aa-105a23312c5e)

**Comment by johnrudell** [comment]
@Dhrumil-Sentry Unfortunately, this if statement isn't available for the `Number of Errors` alert

**Comment by Dhrumil-Sentry** [comment]
@johnrudell this is a valid request and we've added this to our backlog

**Comment by ceorourke** [comment]
@johnrudell If I understand correctly, the request is to support the `assigned:[#team-a, #team-b]` search filter for metric alerts? Let's say it's the number of errors one, it'd only fire when the number of errors across issues in the target project that are assigned to your team(s) has reached the defined threshold?

**Comment by johnrudell** [comment]
@ceorourke Correct. Although I believe the filter is working based on my alert working. The UI just says it's not supported.

**Comment by ceorourke** [comment]
Thanks for the clarification!

**Comment by ceorourke** [comment]
Related issue: https://github.com/getsentry/sentry/issues/66480

**Comment by jangjodi** [comment]
I do not believe that the `assigned` filter is supported in metric alerts currently. I attempted to reproduce this and found that this [alert](https://jodi-sj.sentry.io/alerts/rules/details/240257/) was triggered, whereas this [team alert](https://jodi-sj.sentry.io/alerts/rules/details/165541/) was not when I purposely spammed this [error](https://jodi-sj.sentry.io/issues/5257078216/?query=start%3D2024-04-23T21%3A30%3A11%26end%3D2024-04-30T20%3A05%3A11%26groupStatsPeriod%3Dauto&referrer=alerts-related-issues-issue-stream) that was assigned to the team.

**Comment by rachrwang** [comment]
Thanks for sharing this example @rodolfoBee!
